Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```
@@ -24,66 +47,177 @@ public partial class GestureManager : Singleton<GestureManager>
+
+        private bool HandPressed { get { return pressedHands.Count > 0; } }
+        private HashSet<uint> pressedHands = new HashSet<uint>();
+
+        private InteractionSourceState currentHandState;
 
         void Start()
         {
+            InteractionManager.SourcePressed += InteractionManager_SourcePressed;
+            InteractionManager.SourceReleased += InteractionManager_SourceReleased;
+            InteractionManager.SourceUpdated += InteractionManager_SourceUpdated;
+            InteractionManager.SourceLost += InteractionManager_SourceLost;
+
             // Create a new GestureRecognizer. Sign up for tapped events.
             gestureRecognizer = new GestureRecognizer();
             gestureRecognizer.SetRecognizableGestures(GestureSettings.Tap);
 
+            manipulationRecognizer = new GestureRecognizer();
+            manipulationRecognizer.SetRecognizableGestures(GestureSettings.ManipulationTranslate);
+
             gestureRecognizer.TappedEvent += GestureRecognizer_TappedEvent;
 
+            manipulationRecognizer.ManipulationStartedEvent += ManipulationRecognizer_ManipulationStartedEvent;
+            manipulationRecognizer.ManipulationUpdatedEvent += ManipulationRecognizer_ManipulationUpdatedEvent;
+            manipulationRecognizer.ManipulationCompletedEvent += ManipulationRecognizer_ManipulationCompletedEvent;
+            manipulationRecognizer.ManipulationCanceledEvent += ManipulationRecognizer_ManipulationCanceledEvent;
+
             // Start looking for gestures.
             gestureRecognizer.StartCapturingGestures();
+            manipulationRecognizer.StartCapturingGestures();
         }
 
-        private void OnTap()
+
+
+        private void InteractionManager_SourcePressed(InteractionSourceState state)
+        {
+            if (!HandPressed)
+            {
+                currentHandState = state;
+            }
+
+            pressedHands.Add(state.source.id);
```

Sorry Neeraj, but I'm going to disagree on these "if contains" checks. HashSet.Remove returns true if the item was found and removed. It returns false if the item was not found. An if contains check is actually duplicate work and isn't needed. BTW, pretty much everything that inherits from collection or list silently ignore a request to remove something that isn't there.